### PR TITLE
Integral integration

### PR DESCRIPTION
**Pull request by certara-yaroslavivanov**

### CODE REVIEW

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-datasets/components/import-datasets.component.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ import { UploadedFileViewModel } from 'src/app/projects/models/uploaded-file.vie
 export class ImportDatasetsComponent implements OnInit, OnDestroy {
 
   @ViewChild('fileInput', { static: false }) public fileInput: ElementRef;
-  public hasIntegralApp = false;
+  public hasIntegralApp = true;
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-datasets/components/import-datasets.component.ts:32]: With linting we might have explicit type declarations

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts [hunk 2]
```diff
@@ -1,67 +1,175 @@
...
 @Component({
     selector: 'app-import-integral',
     templateUrl: './import-integral.component.html',
     styleUrls: ['./import-integral.component.scss']
 })
 export class ImportIntegralComponent implements OnInit, OnDestroy {
-    private _fb = new FormBuilder();
-    private _instanceForm: FormGroup;
-    private _data: any[] = [];
+    get selectedRepository() {
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts:19]: Explicit accessor and return type

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts [hunk 3]
```diff
@@ -1,67 +1,175 @@
...
+        return this._selectedRepository;
+    }
+    set selectedRepository(selection: any) {
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts:22]: Explicit accessor and return type

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts [hunk 4]
```diff
@@ -1,67 +1,175 @@
...
+            return;
+        }
 
-        this._dataSource.initialize();
+        this._httpClientService.initializeHttpOptions().then(() => {
+            this._initializeData();
+        });
     }
 
     ngOnDestroy(): void {
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts:69]: Explicit accessor

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/services/repository-manager.service.ts [hunk 5]
```diff
@@ -1,17 +1,60 @@
...
 export class RepositoryManagerService {
-    constructor(private readonly httpClientService: AuthenticatedConfiguredSharedFilesHttpClientService) { }
+    private _integralToken: string;
 
-    getRepositoryItems(folder?: IRepositoryItem): Observable<IRepositoryItem[]> {
-        const folderId = folder ? folder.Id : '';
-        return this.httpClientService.getApiResource<IRepositoryItem[]>(`integral/${folderId}`);
+    constructor(private readonly _httpClientService: AuthenticatedHttpClientService) { }
+
+    hasIntegralToken(): boolean {
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/services/repository-manager.service.ts:17]: accessor

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts [hunk 6]
```diff
@@ -1,67 +1,175 @@
...
+    set selectedRepository(selection: any) {
+        this._selectedRepository = selection;
+        this._loadCurrentRepositoryData();
+    }
 
     private _dataSource: RepositoryItemTreeDataSource;
     private _treeControl: FlatTreeControl<RepositoryItemTreeNode>;
-
-    private _subscription: Subscription;
+    private _favoriteFolders: IRepositoryItem[];
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/components/import-integral.component.ts:29]: explicit type declarations everywhere. lets talk linting precommit

File: src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/services/repository-manager.service.ts [hunk 7]
```diff
@@ -1,17 +1,60 @@
...
+        const headers = new HttpHeaders()
+            .append('Authorization', `Bearer ${this._integralToken}`);
+        const options = { headers };
+        return this._httpClientService.getApiResource<any[]>(`integral/repositoriesforuser`, options);
+    }
+
+    getRepositoryItems(repositoryUuid: string, folder?: IRepositoryItem): Observable<IRepositoryItem[]> {
+        const headers = new HttpHeaders()
+            .append('PMX-RepositoryUuid', repositoryUuid)
+            .append('Authorization', `Bearer ${this._integralToken}`);
```
Comment (ankitsinghcertara) [inline src/AngularWebWinNonlinCloud/ClientApp/src/app/projects/import/import-integral/services/repository-manager.service.ts:47]: Did you ensure that the headers get appended to other headers passed for the http calls like wnl-user-id, wnl-application-id, access-token etc and not replacing them

**Comment by certara-yaroslavivanov** [comment]
@ankitsinghcertara I configured using the same Lint rules as in angular-web-components, also added TsLint & Prettier. They fixed all Lint errors in the files in the PR and formatted them

**Pull request merged**